src/ophyd_async/core/detector.py

```diff
@@ -171,7 +171,10 @@ async def trigger(self) -> None:
         indices_written = await self.writer.get_indices_written()
         written_status = await self.controller.arm(DetectorTrigger.internal, num=1)
         await written_status
-        await self.writer.wait_for_index(indices_written + 1)
+        await asyncio.wait(
+            [self.writer.wait_for_index(indices_written + 1)],
```

Comment:
Should we add `timeout` to `DetectorWriter.wait_for_index` so this line becomes `await self.writer.wait_for_index(indices_written + 1, timeout=DEFAULT_TIMEOUT)`? We already added a `timeout` to `wait_for_value` so this is just a case of passing that value down...

Reply:
Implemented in https://github.com/bluesky/ophyd-async/pull/55